src/Parser/Monk/Brewmaster/Modules/Spells/IronSkinBrew.js

```diff
@@ -21,17 +21,71 @@ class IronSkinBrew extends Analyzer {
   hitsWithoutIronSkinBrew = 0;
   damageWithoutIronSkinBrew = 0;
 
+  // duration tracking to record clipping of buff
+  lastDurationCheck = 0;
+  totalDuration = 0;
+  durationLost = 0;
+  currentDuration = 0;
+  durationPerCast = 6000; // base
+  durationPerPurify = 0;
+  durationCap = -1;
+
+  buffShouldBeApplied = false;
+  buffNeverApplied = true;
+
+  on_initialized() {
```

Comment:
Your additions here seem to be about a different analysis from what the original analyzer did. How about making a new module for it instead? I think keeping things separate and small modules will make maintenance and reuse later easier.